crates/module-system/sov-modules-api/src/transaction/mod.rs

```diff
@@ -229,14 +231,33 @@ impl<R: TransactionCallable, S: Spec> Transaction<R, S> {
         })?;
         serialized_tx.extend_from_slice(chain_hash);
 
-        match &self.versioned_tx {
-            VersionedTx::V0(inner) => {
-                MeteredSignature::new::<S>(inner.signature.clone())
-                    .verify(&inner.pub_key, &serialized_tx, meter)
-                    .map_err(TransactionVerificationError::from)?;
-            }
-        }
-        Ok(())
+        self.verify_inner(&serialized_tx, meter)
+    }
+
+    /// Check whether the transaction has been signed correctly using EIP712.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    ///  * The signature is wrong
+    ///  * Serializing or hashing the transaction fails
+    ///  * Any operation runs out of gas
+    pub fn verify_eip712(
```

Comment:
Shouldn't this method be available only for EvmCryptoSpec, so it can be used exclusively by EIP-712 auth?